static/app/views/performance/transactionSummary/transactionThresholdButton.tsx

```diff
@@ -0,0 +1,157 @@
+  TransactionThresholdMetric,
+} from './transactionThresholdModal';
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  transactionName: string;
+  projects: Project[];
+  eventView: EventView;
+  onChangeThreshold?: (threshold: number, metric: TransactionThresholdMetric) => void;
+};
+
+type State = {
+  transactionThreshold: number | undefined;
+  transactionThresholdMetric: TransactionThresholdMetric | undefined;
+  loadingThreshold: boolean;
+};
+
+class TransactionThresholdButton extends Component<Props, State> {
+  state: State = {
+    transactionThreshold: undefined,
+    transactionThresholdMetric: undefined,
+    loadingThreshold: false,
+  };
+
+  componentDidMount() {
+    this.fetchTransactionThreshold();
+  }
+
+  getProject() {
+    const {projects, eventView} = this.props;
+    if (!defined(eventView)) {
+      return undefined;
+    }
+
+    const projectId = String(eventView.project[0]);
+    const project = projects.find(proj => proj.id === projectId);
+
+    return project;
+  }
+
+  fetchTransactionThreshold = () => {
+    const {api, organization, transactionName} = this.props;
+
+    const project = this.getProject();
+    if (!defined(project)) {
+      return;
+    }
+    const transactionThresholdUrl = `/organizations/${organization.slug}/project-transaction-threshold-override/`;
+
+    this.setState({loadingThreshold: true});
+
+    api
+      .requestPromise(transactionThresholdUrl, {
+        method: 'GET',
+        includeAllArgs: true,
+        query: {
+          project: project.id,
+          transaction: transactionName,
+        },
+      })
+      .then(([data]) => {
+        this.setState({
+          loadingThreshold: false,
+          transactionThreshold: data.threshold,
+          transactionThresholdMetric: data.metric,
+        });
+      })
+      .catch(() => {
+        const projectThresholdUrl = `/projects/${organization.slug}/${project.slug}/transaction-threshold/configure/`;
+        this.props.api
+          .requestPromise(projectThresholdUrl, {
+            method: 'GET',
+            includeAllArgs: true,
+            query: {
+              project: project.id,
+            },
+          })
+          .then(([data]) => {
+            this.setState({
+              loadingThreshold: false,
+              transactionThreshold: data.threshold,
+              transactionThresholdMetric: data.metric,
+            });
+          })
+          .catch(err => {
+            this.setState({loadingThreshold: false});
+            const errorMessage = err.responseJSON?.threshold ?? null;
+            addErrorMessage(errorMessage);
+          });
+      });
+  };
+
+  onChangeThreshold(threshold: number, metric: TransactionThresholdMetric) {
+    const {onChangeThreshold} = this.props;
+    this.setState({
+      transactionThreshold: threshold,
+      transactionThresholdMetric: metric,
+    });
+
+    if (defined(onChangeThreshold)) {
+      onChangeThreshold(threshold, metric);
+    }
```

Comment:
Nit: I personally like to use optional chaining, not a blocker if you leave this alone

```suggestion
    onChangeThreshold?.(threshold, metric);
```